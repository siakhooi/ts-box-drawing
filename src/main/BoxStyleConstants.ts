import {BoxStyle} from './BoxStyle';

const THIN = new BoxStyle(
  '\u2500',
  '\u2502',
  '\u250c',
  '\u2510',
  '\u2514',
  '\u2518'
);
const THICK = new BoxStyle(
  '\u2501',
  '\u2503',
  '\u250F',
  '\u2513',
  '\u2517',
  '\u251B'
);
const DOUBLE = new BoxStyle(
  '\u2550',
  '\u2551',
  '\u2554',
  '\u2557',
  '\u255A',
  '\u255D'
);
const THINCURVE = new BoxStyle(
  '\u2500',
  '\u2502',
  '\u256D',
  '\u256E',
  '\u2570',
  '\u256F'
);

export {THIN, THICK, DOUBLE, THINCURVE};
