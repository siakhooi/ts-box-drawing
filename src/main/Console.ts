import util from 'util';

class ConsoleClass {
  println(format: string, ...texts: string[]): void {
    console.log(util.format(format, ...texts));
  }
}

export const Console = new ConsoleClass();
