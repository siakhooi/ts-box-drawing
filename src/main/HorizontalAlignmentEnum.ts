export enum HorizontalAlignmentEnum {
  LEFT,
  CENTER,
  RIGHT,
}
