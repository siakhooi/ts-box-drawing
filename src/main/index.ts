import {BoxStyle} from './BoxStyle';
import {Console} from './Console';
import {THIN, THICK, DOUBLE, THINCURVE} from './BoxStyleConstants';

function drawBox(boxStyle: BoxStyle, text: string) {
  const w = text.length;
  Console.printf(
    '%s%s%s',
    boxStyle.TOP_LEFT,
    boxStyle.HORIZONTAL.repeat(w),
    boxStyle.TOP_RIGHT
  );
  Console.printf('%s%s%s', boxStyle.VERTICAL, text, boxStyle.VERTICAL);
  Console.printf(
    '%s%s%s',
    boxStyle.BOTTOM_LEFT,
    boxStyle.HORIZONTAL.repeat(w),
    boxStyle.BOTTOM_RIGHT
  );
}
export function drawBoxThin(text: string) {
  drawBox(THIN, text);
}
export function drawBoxThick(text: string) {
  drawBox(THICK, text);
}
export function drawBoxDouble(text: string) {
  drawBox(DOUBLE, text);
}
export function drawBoxThinCurve(text: string) {
  drawBox(THINCURVE, text);
}
