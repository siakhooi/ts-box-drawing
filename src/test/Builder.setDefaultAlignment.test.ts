import {Console} from '../main/Console';
import {BoxDrawingBuilder, HorizontalAlignmentEnum} from '../main/index';
import {
  ALIGN_TEST_DATA,
  ALL_STYLES,
  TestDataUnit,
  TestStyleUnit,
} from './TestData';

describe.each(ALL_STYLES)(
  'setDefaultAlignment()',
  (testStyle: TestStyleUnit) => {
    describe.each([
      HorizontalAlignmentEnum.LEFT,
      HorizontalAlignmentEnum.CENTER,
      HorizontalAlignmentEnum.RIGHT,
    ])('%s', (horizontalAlignment: HorizontalAlignmentEnum) => {
      test.each(ALIGN_TEST_DATA)(
        `${testStyle.title} [$title]`,
        (testData: TestDataUnit) => {
          const allData: string[] = [];
          Console.println = jest
            .fn()
            .mockImplementation(text => allData.push(text));
          new BoxDrawingBuilder()
            .setData(testData.data)
            .setStyle(testStyle.style)
            .setDefaultAlignment(horizontalAlignment)
            .drawBox();
          expect(allData).toMatchSnapshot();
        }
      );
    });
  }
);
