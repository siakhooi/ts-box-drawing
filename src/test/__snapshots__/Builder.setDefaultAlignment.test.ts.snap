// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setDefaultAlignment() Center DOUBLE [Simple 2x2] 1`] = `
[
  "╔════════╦════════╦═════════╦════════╗",
  "║Column 1║Column 2║Column 3A║Column 4║",
  "╠════════╬════════╬═════════╬════════╣",
  "║  Left  ║  Ctr   ║   Ctr   ║ Right  ║",
  "╠════════╬════════╬═════════╬════════╣",
  "║ Left 1 ║  Ctr2  ║  Ctr3   ║Right 4 ║",
  "╚════════╩════════╩═════════╩════════╝",
]
`;

exports[`setDefaultAlignment() Center HORIZONTAL_DOUBLE_VERTICAL_THIN [Simple 2x2] 1`] = `
[
  "╒════════╤════════╤═════════╤════════╕",
  "│Column 1│Column 2│Column 3A│Column 4│",
  "╞════════╪════════╪═════════╪════════╡",
  "│  Left  │  Ctr   │   Ctr   │ Right  │",
  "╞════════╪════════╪═════════╪════════╡",
  "│ Left 1 │  Ctr2  │  Ctr3   │Right 4 │",
  "╘════════╧════════╧═════════╧════════╛",
]
`;

exports[`setDefaultAlignment() Center HORIZONTAL_THICK_VERTICAL_THIN [Simple 2x2] 1`] = `
[
  "┍━━━━━━━━┯━━━━━━━━┯━━━━━━━━━┯━━━━━━━━┑",
  "│Column 1│Column 2│Column 3A│Column 4│",
  "┝━━━━━━━━┿━━━━━━━━┿━━━━━━━━━┿━━━━━━━━┥",
  "│  Left  │  Ctr   │   Ctr   │ Right  │",
  "┝━━━━━━━━┿━━━━━━━━┿━━━━━━━━━┿━━━━━━━━┥",
  "│ Left 1 │  Ctr2  │  Ctr3   │Right 4 │",
  "┕━━━━━━━━┷━━━━━━━━┷━━━━━━━━━┷━━━━━━━━┙",
]
`;

exports[`setDefaultAlignment() Center HORIZONTAL_THIN_VERTICAL_DOUBLE [Simple 2x2] 1`] = `
[
  "╓────────╥────────╥─────────╥────────╖",
  "║Column 1║Column 2║Column 3A║Column 4║",
  "╟────────╫────────╫─────────╫────────╢",
  "║  Left  ║  Ctr   ║   Ctr   ║ Right  ║",
  "╟────────╫────────╫─────────╫────────╢",
  "║ Left 1 ║  Ctr2  ║  Ctr3   ║Right 4 ║",
  "╙────────╨────────╨─────────╨────────╜",
]
`;

exports[`setDefaultAlignment() Center HORIZONTAL_THIN_VERTICAL_THICK [Simple 2x2] 1`] = `
[
  "┎────────┰────────┰─────────┰────────┒",
  "┃Column 1┃Column 2┃Column 3A┃Column 4┃",
  "┠────────╂────────╂─────────╂────────┨",
  "┃  Left  ┃  Ctr   ┃   Ctr   ┃ Right  ┃",
  "┠────────╂────────╂─────────╂────────┨",
  "┃ Left 1 ┃  Ctr2  ┃  Ctr3   ┃Right 4 ┃",
  "┖────────┸────────┸─────────┸────────┚",
]
`;

exports[`setDefaultAlignment() Center INNER_DOUBLE_OUTER_THIN [Simple 2x2] 1`] = `
[
  "┌────────╥────────╥─────────╥────────┐",
  "│Column 1║Column 2║Column 3A║Column 4│",
  "╞════════╬════════╬═════════╬════════╡",
  "│  Left  ║  Ctr   ║   Ctr   ║ Right  │",
  "╞════════╬════════╬═════════╬════════╡",
  "│ Left 1 ║  Ctr2  ║  Ctr3   ║Right 4 │",
  "└────────╨────────╨─────────╨────────┘",
]
`;

exports[`setDefaultAlignment() Center INNER_DOUBLE_OUTER_THIN_CURVE [Simple 2x2] 1`] = `
[
  "╭────────╥────────╥─────────╥────────╮",
  "│Column 1║Column 2║Column 3A║Column 4│",
  "╞════════╬════════╬═════════╬════════╡",
  "│  Left  ║  Ctr   ║   Ctr   ║ Right  │",
  "╞════════╬════════╬═════════╬════════╡",
  "│ Left 1 ║  Ctr2  ║  Ctr3   ║Right 4 │",
  "╰────────╨────────╨─────────╨────────╯",
]
`;

exports[`setDefaultAlignment() Center INNER_THICK_OUTER_THIN [Simple 2x2] 1`] = `
[
  "┌────────┰────────┰─────────┰────────┐",
  "│Column 1┃Column 2┃Column 3A┃Column 4│",
  "┝━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┥",
  "│  Left  ┃  Ctr   ┃   Ctr   ┃ Right  │",
  "┝━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┥",
  "│ Left 1 ┃  Ctr2  ┃  Ctr3   ┃Right 4 │",
  "└────────┸────────┸─────────┸────────┘",
]
`;

exports[`setDefaultAlignment() Center INNER_THICK_OUTER_THIN_CURVE [Simple 2x2] 1`] = `
[
  "╭────────┰────────┰─────────┰────────╮",
  "│Column 1┃Column 2┃Column 3A┃Column 4│",
  "┝━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┥",
  "│  Left  ┃  Ctr   ┃   Ctr   ┃ Right  │",
  "┝━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┥",
  "│ Left 1 ┃  Ctr2  ┃  Ctr3   ┃Right 4 │",
  "╰────────┸────────┸─────────┸────────╯",
]
`;

exports[`setDefaultAlignment() Center INNER_THIN_OUTER_DOUBLE [Simple 2x2] 1`] = `
[
  "╔════════╤════════╤═════════╤════════╗",
  "║Column 1│Column 2│Column 3A│Column 4║",
  "╟────────┼────────┼─────────┼────────╢",
  "║  Left  │  Ctr   │   Ctr   │ Right  ║",
  "╟────────┼────────┼─────────┼────────╢",
  "║ Left 1 │  Ctr2  │  Ctr3   │Right 4 ║",
  "╚════════╧════════╧═════════╧════════╝",
]
`;

exports[`setDefaultAlignment() Center INNER_THIN_OUTER_THICK [Simple 2x2] 1`] = `
[
  "┏━━━━━━━━┯━━━━━━━━┯━━━━━━━━━┯━━━━━━━━┓",
  "┃Column 1│Column 2│Column 3A│Column 4┃",
  "┠────────┼────────┼─────────┼────────┨",
  "┃  Left  │  Ctr   │   Ctr   │ Right  ┃",
  "┠────────┼────────┼─────────┼────────┨",
  "┃ Left 1 │  Ctr2  │  Ctr3   │Right 4 ┃",
  "┗━━━━━━━━┷━━━━━━━━┷━━━━━━━━━┷━━━━━━━━┛",
]
`;

exports[`setDefaultAlignment() Center THICK [Simple 2x2] 1`] = `
[
  "┏━━━━━━━━┳━━━━━━━━┳━━━━━━━━━┳━━━━━━━━┓",
  "┃Column 1┃Column 2┃Column 3A┃Column 4┃",
  "┣━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┫",
  "┃  Left  ┃  Ctr   ┃   Ctr   ┃ Right  ┃",
  "┣━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┫",
  "┃ Left 1 ┃  Ctr2  ┃  Ctr3   ┃Right 4 ┃",
  "┗━━━━━━━━┻━━━━━━━━┻━━━━━━━━━┻━━━━━━━━┛",
]
`;

exports[`setDefaultAlignment() Center THIN [Simple 2x2] 1`] = `
[
  "┌────────┬────────┬─────────┬────────┐",
  "│Column 1│Column 2│Column 3A│Column 4│",
  "├────────┼────────┼─────────┼────────┤",
  "│  Left  │  Ctr   │   Ctr   │ Right  │",
  "├────────┼────────┼─────────┼────────┤",
  "│ Left 1 │  Ctr2  │  Ctr3   │Right 4 │",
  "└────────┴────────┴─────────┴────────┘",
]
`;

exports[`setDefaultAlignment() Center THINCURVE [Simple 2x2] 1`] = `
[
  "╭────────┬────────┬─────────┬────────╮",
  "│Column 1│Column 2│Column 3A│Column 4│",
  "├────────┼────────┼─────────┼────────┤",
  "│  Left  │  Ctr   │   Ctr   │ Right  │",
  "├────────┼────────┼─────────┼────────┤",
  "│ Left 1 │  Ctr2  │  Ctr3   │Right 4 │",
  "╰────────┴────────┴─────────┴────────╯",
]
`;

exports[`setDefaultAlignment() Center VERTICALBAR_PLUS_EQUALS [Simple 2x2] 1`] = `
[
  "+========+========+=========+========+",
  "|Column 1|Column 2|Column 3A|Column 4|",
  "+========+========+=========+========+",
  "|  Left  |  Ctr   |   Ctr   | Right  |",
  "+========+========+=========+========+",
  "| Left 1 |  Ctr2  |  Ctr3   |Right 4 |",
  "+========+========+=========+========+",
]
`;

exports[`setDefaultAlignment() Center VERTICALBAR_PLUS_MINUS [Simple 2x2] 1`] = `
[
  "+--------+--------+---------+--------+",
  "|Column 1|Column 2|Column 3A|Column 4|",
  "+--------+--------+---------+--------+",
  "|  Left  |  Ctr   |   Ctr   | Right  |",
  "+--------+--------+---------+--------+",
  "| Left 1 |  Ctr2  |  Ctr3   |Right 4 |",
  "+--------+--------+---------+--------+",
]
`;

exports[`setDefaultAlignment() Center VERTICALBAR_PLUS_MINUS_SLASHES [Simple 2x2] 1`] = `
[
  "/--------+--------+---------+--------\\",
  "|Column 1|Column 2|Column 3A|Column 4|",
  "+--------+--------+---------+--------+",
  "|  Left  |  Ctr   |   Ctr   | Right  |",
  "+--------+--------+---------+--------+",
  "| Left 1 |  Ctr2  |  Ctr3   |Right 4 |",
  "\\--------+--------+---------+--------/",
]
`;

exports[`setDefaultAlignment() Left DOUBLE [Simple 2x2] 1`] = `
[
  "╔════════╦════════╦═════════╦════════╗",
  "║Column 1║Column 2║Column 3A║Column 4║",
  "╠════════╬════════╬═════════╬════════╣",
  "║Left    ║Ctr     ║Ctr      ║Right   ║",
  "╠════════╬════════╬═════════╬════════╣",
  "║Left 1  ║Ctr2    ║Ctr3     ║Right 4 ║",
  "╚════════╩════════╩═════════╩════════╝",
]
`;

exports[`setDefaultAlignment() Left HORIZONTAL_DOUBLE_VERTICAL_THIN [Simple 2x2] 1`] = `
[
  "╒════════╤════════╤═════════╤════════╕",
  "│Column 1│Column 2│Column 3A│Column 4│",
  "╞════════╪════════╪═════════╪════════╡",
  "│Left    │Ctr     │Ctr      │Right   │",
  "╞════════╪════════╪═════════╪════════╡",
  "│Left 1  │Ctr2    │Ctr3     │Right 4 │",
  "╘════════╧════════╧═════════╧════════╛",
]
`;

exports[`setDefaultAlignment() Left HORIZONTAL_THICK_VERTICAL_THIN [Simple 2x2] 1`] = `
[
  "┍━━━━━━━━┯━━━━━━━━┯━━━━━━━━━┯━━━━━━━━┑",
  "│Column 1│Column 2│Column 3A│Column 4│",
  "┝━━━━━━━━┿━━━━━━━━┿━━━━━━━━━┿━━━━━━━━┥",
  "│Left    │Ctr     │Ctr      │Right   │",
  "┝━━━━━━━━┿━━━━━━━━┿━━━━━━━━━┿━━━━━━━━┥",
  "│Left 1  │Ctr2    │Ctr3     │Right 4 │",
  "┕━━━━━━━━┷━━━━━━━━┷━━━━━━━━━┷━━━━━━━━┙",
]
`;

exports[`setDefaultAlignment() Left HORIZONTAL_THIN_VERTICAL_DOUBLE [Simple 2x2] 1`] = `
[
  "╓────────╥────────╥─────────╥────────╖",
  "║Column 1║Column 2║Column 3A║Column 4║",
  "╟────────╫────────╫─────────╫────────╢",
  "║Left    ║Ctr     ║Ctr      ║Right   ║",
  "╟────────╫────────╫─────────╫────────╢",
  "║Left 1  ║Ctr2    ║Ctr3     ║Right 4 ║",
  "╙────────╨────────╨─────────╨────────╜",
]
`;

exports[`setDefaultAlignment() Left HORIZONTAL_THIN_VERTICAL_THICK [Simple 2x2] 1`] = `
[
  "┎────────┰────────┰─────────┰────────┒",
  "┃Column 1┃Column 2┃Column 3A┃Column 4┃",
  "┠────────╂────────╂─────────╂────────┨",
  "┃Left    ┃Ctr     ┃Ctr      ┃Right   ┃",
  "┠────────╂────────╂─────────╂────────┨",
  "┃Left 1  ┃Ctr2    ┃Ctr3     ┃Right 4 ┃",
  "┖────────┸────────┸─────────┸────────┚",
]
`;

exports[`setDefaultAlignment() Left INNER_DOUBLE_OUTER_THIN [Simple 2x2] 1`] = `
[
  "┌────────╥────────╥─────────╥────────┐",
  "│Column 1║Column 2║Column 3A║Column 4│",
  "╞════════╬════════╬═════════╬════════╡",
  "│Left    ║Ctr     ║Ctr      ║Right   │",
  "╞════════╬════════╬═════════╬════════╡",
  "│Left 1  ║Ctr2    ║Ctr3     ║Right 4 │",
  "└────────╨────────╨─────────╨────────┘",
]
`;

exports[`setDefaultAlignment() Left INNER_DOUBLE_OUTER_THIN_CURVE [Simple 2x2] 1`] = `
[
  "╭────────╥────────╥─────────╥────────╮",
  "│Column 1║Column 2║Column 3A║Column 4│",
  "╞════════╬════════╬═════════╬════════╡",
  "│Left    ║Ctr     ║Ctr      ║Right   │",
  "╞════════╬════════╬═════════╬════════╡",
  "│Left 1  ║Ctr2    ║Ctr3     ║Right 4 │",
  "╰────────╨────────╨─────────╨────────╯",
]
`;

exports[`setDefaultAlignment() Left INNER_THICK_OUTER_THIN [Simple 2x2] 1`] = `
[
  "┌────────┰────────┰─────────┰────────┐",
  "│Column 1┃Column 2┃Column 3A┃Column 4│",
  "┝━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┥",
  "│Left    ┃Ctr     ┃Ctr      ┃Right   │",
  "┝━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┥",
  "│Left 1  ┃Ctr2    ┃Ctr3     ┃Right 4 │",
  "└────────┸────────┸─────────┸────────┘",
]
`;

exports[`setDefaultAlignment() Left INNER_THICK_OUTER_THIN_CURVE [Simple 2x2] 1`] = `
[
  "╭────────┰────────┰─────────┰────────╮",
  "│Column 1┃Column 2┃Column 3A┃Column 4│",
  "┝━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┥",
  "│Left    ┃Ctr     ┃Ctr      ┃Right   │",
  "┝━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┥",
  "│Left 1  ┃Ctr2    ┃Ctr3     ┃Right 4 │",
  "╰────────┸────────┸─────────┸────────╯",
]
`;

exports[`setDefaultAlignment() Left INNER_THIN_OUTER_DOUBLE [Simple 2x2] 1`] = `
[
  "╔════════╤════════╤═════════╤════════╗",
  "║Column 1│Column 2│Column 3A│Column 4║",
  "╟────────┼────────┼─────────┼────────╢",
  "║Left    │Ctr     │Ctr      │Right   ║",
  "╟────────┼────────┼─────────┼────────╢",
  "║Left 1  │Ctr2    │Ctr3     │Right 4 ║",
  "╚════════╧════════╧═════════╧════════╝",
]
`;

exports[`setDefaultAlignment() Left INNER_THIN_OUTER_THICK [Simple 2x2] 1`] = `
[
  "┏━━━━━━━━┯━━━━━━━━┯━━━━━━━━━┯━━━━━━━━┓",
  "┃Column 1│Column 2│Column 3A│Column 4┃",
  "┠────────┼────────┼─────────┼────────┨",
  "┃Left    │Ctr     │Ctr      │Right   ┃",
  "┠────────┼────────┼─────────┼────────┨",
  "┃Left 1  │Ctr2    │Ctr3     │Right 4 ┃",
  "┗━━━━━━━━┷━━━━━━━━┷━━━━━━━━━┷━━━━━━━━┛",
]
`;

exports[`setDefaultAlignment() Left THICK [Simple 2x2] 1`] = `
[
  "┏━━━━━━━━┳━━━━━━━━┳━━━━━━━━━┳━━━━━━━━┓",
  "┃Column 1┃Column 2┃Column 3A┃Column 4┃",
  "┣━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┫",
  "┃Left    ┃Ctr     ┃Ctr      ┃Right   ┃",
  "┣━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┫",
  "┃Left 1  ┃Ctr2    ┃Ctr3     ┃Right 4 ┃",
  "┗━━━━━━━━┻━━━━━━━━┻━━━━━━━━━┻━━━━━━━━┛",
]
`;

exports[`setDefaultAlignment() Left THIN [Simple 2x2] 1`] = `
[
  "┌────────┬────────┬─────────┬────────┐",
  "│Column 1│Column 2│Column 3A│Column 4│",
  "├────────┼────────┼─────────┼────────┤",
  "│Left    │Ctr     │Ctr      │Right   │",
  "├────────┼────────┼─────────┼────────┤",
  "│Left 1  │Ctr2    │Ctr3     │Right 4 │",
  "└────────┴────────┴─────────┴────────┘",
]
`;

exports[`setDefaultAlignment() Left THINCURVE [Simple 2x2] 1`] = `
[
  "╭────────┬────────┬─────────┬────────╮",
  "│Column 1│Column 2│Column 3A│Column 4│",
  "├────────┼────────┼─────────┼────────┤",
  "│Left    │Ctr     │Ctr      │Right   │",
  "├────────┼────────┼─────────┼────────┤",
  "│Left 1  │Ctr2    │Ctr3     │Right 4 │",
  "╰────────┴────────┴─────────┴────────╯",
]
`;

exports[`setDefaultAlignment() Left VERTICALBAR_PLUS_EQUALS [Simple 2x2] 1`] = `
[
  "+========+========+=========+========+",
  "|Column 1|Column 2|Column 3A|Column 4|",
  "+========+========+=========+========+",
  "|Left    |Ctr     |Ctr      |Right   |",
  "+========+========+=========+========+",
  "|Left 1  |Ctr2    |Ctr3     |Right 4 |",
  "+========+========+=========+========+",
]
`;

exports[`setDefaultAlignment() Left VERTICALBAR_PLUS_MINUS [Simple 2x2] 1`] = `
[
  "+--------+--------+---------+--------+",
  "|Column 1|Column 2|Column 3A|Column 4|",
  "+--------+--------+---------+--------+",
  "|Left    |Ctr     |Ctr      |Right   |",
  "+--------+--------+---------+--------+",
  "|Left 1  |Ctr2    |Ctr3     |Right 4 |",
  "+--------+--------+---------+--------+",
]
`;

exports[`setDefaultAlignment() Left VERTICALBAR_PLUS_MINUS_SLASHES [Simple 2x2] 1`] = `
[
  "/--------+--------+---------+--------\\",
  "|Column 1|Column 2|Column 3A|Column 4|",
  "+--------+--------+---------+--------+",
  "|Left    |Ctr     |Ctr      |Right   |",
  "+--------+--------+---------+--------+",
  "|Left 1  |Ctr2    |Ctr3     |Right 4 |",
  "\\--------+--------+---------+--------/",
]
`;

exports[`setDefaultAlignment() Right DOUBLE [Simple 2x2] 1`] = `
[
  "╔════════╦════════╦═════════╦════════╗",
  "║Column 1║Column 2║Column 3A║Column 4║",
  "╠════════╬════════╬═════════╬════════╣",
  "║    Left║     Ctr║      Ctr║   Right║",
  "╠════════╬════════╬═════════╬════════╣",
  "║  Left 1║    Ctr2║     Ctr3║ Right 4║",
  "╚════════╩════════╩═════════╩════════╝",
]
`;

exports[`setDefaultAlignment() Right HORIZONTAL_DOUBLE_VERTICAL_THIN [Simple 2x2] 1`] = `
[
  "╒════════╤════════╤═════════╤════════╕",
  "│Column 1│Column 2│Column 3A│Column 4│",
  "╞════════╪════════╪═════════╪════════╡",
  "│    Left│     Ctr│      Ctr│   Right│",
  "╞════════╪════════╪═════════╪════════╡",
  "│  Left 1│    Ctr2│     Ctr3│ Right 4│",
  "╘════════╧════════╧═════════╧════════╛",
]
`;

exports[`setDefaultAlignment() Right HORIZONTAL_THICK_VERTICAL_THIN [Simple 2x2] 1`] = `
[
  "┍━━━━━━━━┯━━━━━━━━┯━━━━━━━━━┯━━━━━━━━┑",
  "│Column 1│Column 2│Column 3A│Column 4│",
  "┝━━━━━━━━┿━━━━━━━━┿━━━━━━━━━┿━━━━━━━━┥",
  "│    Left│     Ctr│      Ctr│   Right│",
  "┝━━━━━━━━┿━━━━━━━━┿━━━━━━━━━┿━━━━━━━━┥",
  "│  Left 1│    Ctr2│     Ctr3│ Right 4│",
  "┕━━━━━━━━┷━━━━━━━━┷━━━━━━━━━┷━━━━━━━━┙",
]
`;

exports[`setDefaultAlignment() Right HORIZONTAL_THIN_VERTICAL_DOUBLE [Simple 2x2] 1`] = `
[
  "╓────────╥────────╥─────────╥────────╖",
  "║Column 1║Column 2║Column 3A║Column 4║",
  "╟────────╫────────╫─────────╫────────╢",
  "║    Left║     Ctr║      Ctr║   Right║",
  "╟────────╫────────╫─────────╫────────╢",
  "║  Left 1║    Ctr2║     Ctr3║ Right 4║",
  "╙────────╨────────╨─────────╨────────╜",
]
`;

exports[`setDefaultAlignment() Right HORIZONTAL_THIN_VERTICAL_THICK [Simple 2x2] 1`] = `
[
  "┎────────┰────────┰─────────┰────────┒",
  "┃Column 1┃Column 2┃Column 3A┃Column 4┃",
  "┠────────╂────────╂─────────╂────────┨",
  "┃    Left┃     Ctr┃      Ctr┃   Right┃",
  "┠────────╂────────╂─────────╂────────┨",
  "┃  Left 1┃    Ctr2┃     Ctr3┃ Right 4┃",
  "┖────────┸────────┸─────────┸────────┚",
]
`;

exports[`setDefaultAlignment() Right INNER_DOUBLE_OUTER_THIN [Simple 2x2] 1`] = `
[
  "┌────────╥────────╥─────────╥────────┐",
  "│Column 1║Column 2║Column 3A║Column 4│",
  "╞════════╬════════╬═════════╬════════╡",
  "│    Left║     Ctr║      Ctr║   Right│",
  "╞════════╬════════╬═════════╬════════╡",
  "│  Left 1║    Ctr2║     Ctr3║ Right 4│",
  "└────────╨────────╨─────────╨────────┘",
]
`;

exports[`setDefaultAlignment() Right INNER_DOUBLE_OUTER_THIN_CURVE [Simple 2x2] 1`] = `
[
  "╭────────╥────────╥─────────╥────────╮",
  "│Column 1║Column 2║Column 3A║Column 4│",
  "╞════════╬════════╬═════════╬════════╡",
  "│    Left║     Ctr║      Ctr║   Right│",
  "╞════════╬════════╬═════════╬════════╡",
  "│  Left 1║    Ctr2║     Ctr3║ Right 4│",
  "╰────────╨────────╨─────────╨────────╯",
]
`;

exports[`setDefaultAlignment() Right INNER_THICK_OUTER_THIN [Simple 2x2] 1`] = `
[
  "┌────────┰────────┰─────────┰────────┐",
  "│Column 1┃Column 2┃Column 3A┃Column 4│",
  "┝━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┥",
  "│    Left┃     Ctr┃      Ctr┃   Right│",
  "┝━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┥",
  "│  Left 1┃    Ctr2┃     Ctr3┃ Right 4│",
  "└────────┸────────┸─────────┸────────┘",
]
`;

exports[`setDefaultAlignment() Right INNER_THICK_OUTER_THIN_CURVE [Simple 2x2] 1`] = `
[
  "╭────────┰────────┰─────────┰────────╮",
  "│Column 1┃Column 2┃Column 3A┃Column 4│",
  "┝━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┥",
  "│    Left┃     Ctr┃      Ctr┃   Right│",
  "┝━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┥",
  "│  Left 1┃    Ctr2┃     Ctr3┃ Right 4│",
  "╰────────┸────────┸─────────┸────────╯",
]
`;

exports[`setDefaultAlignment() Right INNER_THIN_OUTER_DOUBLE [Simple 2x2] 1`] = `
[
  "╔════════╤════════╤═════════╤════════╗",
  "║Column 1│Column 2│Column 3A│Column 4║",
  "╟────────┼────────┼─────────┼────────╢",
  "║    Left│     Ctr│      Ctr│   Right║",
  "╟────────┼────────┼─────────┼────────╢",
  "║  Left 1│    Ctr2│     Ctr3│ Right 4║",
  "╚════════╧════════╧═════════╧════════╝",
]
`;

exports[`setDefaultAlignment() Right INNER_THIN_OUTER_THICK [Simple 2x2] 1`] = `
[
  "┏━━━━━━━━┯━━━━━━━━┯━━━━━━━━━┯━━━━━━━━┓",
  "┃Column 1│Column 2│Column 3A│Column 4┃",
  "┠────────┼────────┼─────────┼────────┨",
  "┃    Left│     Ctr│      Ctr│   Right┃",
  "┠────────┼────────┼─────────┼────────┨",
  "┃  Left 1│    Ctr2│     Ctr3│ Right 4┃",
  "┗━━━━━━━━┷━━━━━━━━┷━━━━━━━━━┷━━━━━━━━┛",
]
`;

exports[`setDefaultAlignment() Right THICK [Simple 2x2] 1`] = `
[
  "┏━━━━━━━━┳━━━━━━━━┳━━━━━━━━━┳━━━━━━━━┓",
  "┃Column 1┃Column 2┃Column 3A┃Column 4┃",
  "┣━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┫",
  "┃    Left┃     Ctr┃      Ctr┃   Right┃",
  "┣━━━━━━━━╋━━━━━━━━╋━━━━━━━━━╋━━━━━━━━┫",
  "┃  Left 1┃    Ctr2┃     Ctr3┃ Right 4┃",
  "┗━━━━━━━━┻━━━━━━━━┻━━━━━━━━━┻━━━━━━━━┛",
]
`;

exports[`setDefaultAlignment() Right THIN [Simple 2x2] 1`] = `
[
  "┌────────┬────────┬─────────┬────────┐",
  "│Column 1│Column 2│Column 3A│Column 4│",
  "├────────┼────────┼─────────┼────────┤",
  "│    Left│     Ctr│      Ctr│   Right│",
  "├────────┼────────┼─────────┼────────┤",
  "│  Left 1│    Ctr2│     Ctr3│ Right 4│",
  "└────────┴────────┴─────────┴────────┘",
]
`;

exports[`setDefaultAlignment() Right THINCURVE [Simple 2x2] 1`] = `
[
  "╭────────┬────────┬─────────┬────────╮",
  "│Column 1│Column 2│Column 3A│Column 4│",
  "├────────┼────────┼─────────┼────────┤",
  "│    Left│     Ctr│      Ctr│   Right│",
  "├────────┼────────┼─────────┼────────┤",
  "│  Left 1│    Ctr2│     Ctr3│ Right 4│",
  "╰────────┴────────┴─────────┴────────╯",
]
`;

exports[`setDefaultAlignment() Right VERTICALBAR_PLUS_EQUALS [Simple 2x2] 1`] = `
[
  "+========+========+=========+========+",
  "|Column 1|Column 2|Column 3A|Column 4|",
  "+========+========+=========+========+",
  "|    Left|     Ctr|      Ctr|   Right|",
  "+========+========+=========+========+",
  "|  Left 1|    Ctr2|     Ctr3| Right 4|",
  "+========+========+=========+========+",
]
`;

exports[`setDefaultAlignment() Right VERTICALBAR_PLUS_MINUS [Simple 2x2] 1`] = `
[
  "+--------+--------+---------+--------+",
  "|Column 1|Column 2|Column 3A|Column 4|",
  "+--------+--------+---------+--------+",
  "|    Left|     Ctr|      Ctr|   Right|",
  "+--------+--------+---------+--------+",
  "|  Left 1|    Ctr2|     Ctr3| Right 4|",
  "+--------+--------+---------+--------+",
]
`;

exports[`setDefaultAlignment() Right VERTICALBAR_PLUS_MINUS_SLASHES [Simple 2x2] 1`] = `
[
  "/--------+--------+---------+--------\\",
  "|Column 1|Column 2|Column 3A|Column 4|",
  "+--------+--------+---------+--------+",
  "|    Left|     Ctr|      Ctr|   Right|",
  "+--------+--------+---------+--------+",
  "|  Left 1|    Ctr2|     Ctr3| Right 4|",
  "\\--------+--------+---------+--------/",
]
`;
